### PR TITLE
Allow to skip certain JS-tests under `tests/jerry/` and `tests/jerry-…

### PR DESCRIPTION
**Pull request by bzsolt**
…test-suite/`

Add `--skip-list` option to tools/run-tests.py.

Additional modifications:
 - The test `parser-oom.js` doesn't work when system-allocator is enabled,
   hence we skip it in these jobs.
 - The sanitizer-tasks become mandatory.

JerryScript-DCO-1.0-Signed-off-by: Zsolt Borbély zsborbely.u-szeged@partner.samsung.com

### CODE REVIEW

**Comment by zherczeg** [comment]
Shouldn't we skip es2015 tests?

**Comment by bzsolt** [comment]
I don't know any reason to skip those tests.

**Comment by zherczeg** [comment]
Ah, so we always test with all features?

**Comment by bzsolt** [comment]
Actually yes, the tested combinations listed in [tools/run-tests.py](https://github.com/jerryscript-project/jerryscript/blob/master/tools/run-tests.py#L86)

**Comment by jiangzidong** [comment]
Do you mean we should always run ed2015-subset jerry for --jerry-tests?

**Comment by bzsolt** [comment]
The `--jerry-test-suite` option runs the es2015-subset, because these tests are under the `tests/jerry-test-suite/`. If the profile isn't specified, the `es5.1` is the default.

File: tools/runners/run-test-suite.sh [hunk 1]
```diff
@@ -64,6 +71,12 @@ else
     exit 1
 fi
 
+# Remove the skipped tests from list
+for test in "${SKIP_LIST[@]}"
```
Comment (LaszloLango) [inline tools/runners/run-test-suite.sh]: Please use uppercase variable names in bash scripts. (`TEST`)

File: tools/runners/run-test-suite.sh [hunk 2]
```diff
@@ -64,6 +71,12 @@ else
...
+do
+    ( sed -i "/$test/d" $TEST_FILES )
+done
+
 total=$(cat $TEST_FILES | wc -l)
```
Comment (LaszloLango) [inline tools/runners/run-test-suite.sh]: Could you fix this too if you are there? :)

**Comment by jiangzidong** [comment]
Could you let --skip-list to support a list file, like the profile-list in the jerry-test-suite, ? Then I can check in the `run-tests.py`, if the profile is NOT es2015, then I will add the cmd arg: `--skip-list es2015-test-list`

**Comment by zherczeg** [comment]
Ok for me, but at some point we should implement a common test selection system.

**Comment by bzsolt** [comment]
@LaszloLango I've updated the PR.
@jiangzidong So the `--skip-list` should support the 'single-file' and a 'list-based' skip too?
It's possible, if necessary, currently you can skip an entire folder under `tests/jerry-test-suite` (or `tests/jerry`) by
`tools/run-tests.py --jerry-test-suite --skip-list='es2015\/'`.

**Comment by jiangzidong** [comment]
OK, it is good to support skipping a sub folder. LGTM :)

**Pull request merged**